### PR TITLE
Use StartServiceCtrlDispatcherA for the daemon on Windows

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-main-windows.c [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+	/* parse debugging args */
+	g_option_context_add_group(context, fu_debug_get_option_group());
+	if (!g_option_context_parse(context, &argc, &argv, &error)) {
+		g_printerr("Failed to parse command line: %s\n", error->message);
+		return EXIT_FAILURE;
+	}
+
+	/* set up the daemon, which includes coldplugging devices -- then run it */
+	if (!fu_daemon_setup(daemon, &error)) {
```
Comment (superm1) [inline src/fu-main-windows.c:115]: I wonder if you want to pass in here something about what socket to listen on. It's less ifdef code further on then.

Comment (hughsie) [reply src/fu-main-windows.c]: Like that^^?

Comment (superm1) [reply src/fu-main-windows.c]: Yeap that looks good

File: src/fu-daemon.c [hunk 2]
```diff
@@ -2110,11 +2110,9 @@ fu_daemon_set_machine_kind(FuDaemon *self, FuDaemonMachineKind machine_kind)
 }
 
 gboolean
-fu_daemon_setup(FuDaemon *self, GError **error)
+fu_daemon_setup(FuDaemon *self, const gchar *socket_address, GError **error)
 {
 	const gchar *machine_kind = g_getenv("FWUPD_MACHINE_KIND");
-	const gchar *socket_filename = g_getenv("FWUPD_DBUS_SOCKET");
```
Comment (superm1) [inline src/fu-daemon.c:2116]: I think this environment variable should be dropped in matching readme.

Also the client should probably use the same define.

Comment (hughsie) [reply src/fu-daemon.c:2116]: I think we still want it in Linux for the socket self tests, no?

Comment (superm1) [reply src/fu-daemon.c:2116]: Oh I guess still want to use unix socket in Linux, right ok.

**Pull request merged**